android/src/main/java/ie/gov/tracing/network/Fetcher.kt

```diff
@@ -46,8 +63,138 @@ class Fetcher {
+            var trustManager: X509TrustManager? = null
+
+            val cf: CertificateFactory = CertificateFactory.getInstance("X.509")
+            val keyStoreType: String = KeyStore.getDefaultType()
+            val keyStore: KeyStore = KeyStore.getInstance(keyStoreType)
+            keyStore.load(null, null)
+            for (i in certs.indices) {
+                val filename = certs[i]
+                val caInput: InputStream = BufferedInputStream(Fetcher::class.java.getClassLoader()?.getResourceAsStream("assets/$filename.cer"))
+                var ca: Certificate
+                ca = caInput.use { caInput ->
+                    cf.generateCertificate(caInput)
+                }
+                keyStore.setCertificateEntry(filename, ca)
+            }
+            val tmfAlgorithm: String = TrustManagerFactory.getDefaultAlgorithm()
+            val tmf: TrustManagerFactory = TrustManagerFactory.getInstance(tmfAlgorithm)
+            tmf.init(keyStore)
+            val trustManagers: Array<TrustManager> = tmf.getTrustManagers()
+            check(!(trustManagers.size != 1 || trustManagers[0] !is X509TrustManager)) { "Unexpected default trust managers:" + Arrays.toString(trustManagers) }
+            trustManager = trustManagers[0] as X509TrustManager
+            return trustManager
+
+        }
+
+        @JvmStatic
+        fun refreshAuthToken(context: Context): String? {
+            try {
+
+                val url = getURL(REFRESH, context)
+
+                var pin = true
+                var authenticate = true
+
+                val client = getOkClient(pin, authenticate, context)
+                val request = Request.Builder()
+                        .url(url)
+                        .post("".toRequestBody())
+                        .build()
+
+                client.newCall(request).execute().use { response ->
+                    if (!response.isSuccessful) return null
+
+                    val data = response.body!!.string()
+
+                    if (data.isEmpty()) {
+                        return null
+                    }
+
+                    val tokenClass = Gson().fromJson(data, Token::class.java)
+                    if (tokenClass != null) {
+                        SharedPrefs.setString("authToken", tokenClass.token, context)
+                    }
+                    return tokenClass.token
+                }
+
+            } catch (ex: Exception) {
+                Events.raiseError("refresh token error", ex)
+            }
+            return null
+        }
+
+        fun getOkClient(pin: Boolean = true, authenticate: Boolean = true, context: Context): OkHttpClient {
+
+            val authorizationInterceptor = getAuthorizationInterceptor(context)
+            val bearerAuthenticator = getBearerAuthenticator(context)
+            val builder = OkHttpClient.Builder()
+            var usePinning = pin;
+
+            if (BuildConfig.DEBUG) {
```

Comment:
Open to suggestions on what we want to do in debug, rn-ssl-pinning adds HttpLoggingInterceptor so you can see what's going on, setting the user agent makes it easier to distinguish between the module and the actual app on the server. Might make sense to make pinning configurable?

Reply:
Lets add option in the config method to allow it so be disabled. We removed that DEBUG config so would 't use that